core/src/main/java/com/amplitude/core/IdentityStore.kt

```diff
@@ -0,0 +1,178 @@
+                    val action = actionEntry.key
+                    val properties = actionEntry.value
+                    when (action) {
+                        ID_OP_SET -> {
+                            actingProperties.putAll(properties)
+                        }
+                        ID_OP_UNSET -> {
+                            for (entry in properties.entries) {
+                                actingProperties.remove(entry.key)
+                            }
+                        }
+                        ID_OP_SET_ONCE -> {
+                            for (entry in properties.entries) {
+                                actingProperties.getOrPut(entry.key) {
+                                    entry.value
+                                }
+                            }
+                        }
+                        ID_OP_ADD -> {
+                            for (entry in properties.entries) {
+                                val value = entry.value
+                                val actingValue = actingProperties[entry.key] ?: 0
+                                // All lesser numbers can be represented as doubles for simplicity
+                                // TODO: Float values will lose precision when converted to double.
+                                // The current move from (Float | Double) -> BigInteger from org.json
+                                // makes this check unnecessary since all floats will be precisely
+                                // converted.
+                                if (value is Number && actingValue is Number) {
+                                    actingProperties[entry.key] = value.toDouble() + actingValue.toDouble()
+                                }
+                            }
+                        }
+                        ID_OP_APPEND -> {
+                            for (entry in properties.entries) {
+                                val actingValue = actingProperties[entry.key]
+                                val value = entry.value
+                                if (value is List<Any?> && actingValue is List<Any?>) {
+                                    actingProperties[entry.key] = actingValue + value.toMutableList()
+                                }
+                            }
+                        }
+                        ID_OP_PREPEND -> {
+                            for (entry in properties.entries) {
+                                val actingValue = actingProperties[entry.key]
+                                val value = entry.value
+                                if (value is List<Any?> && actingValue is List<Any?>) {
+                                    actingProperties[entry.key] = value.toMutableList() + actingValue
+                                }
+                            }
+                        }
+                        ID_OP_CLEAR_ALL -> {
+                            actingProperties.clear()
+                        }
+
```

Comment:
Maybe add some error detection here incase the action is not recognized.

Reply:
I've change this to no longer support all non-idempotent operations. So we would want to simply do nothing when we see unsupported operations.